### PR TITLE
chore: trim empty lines from templates

### PR DESCRIPTION
**Pull request by sylr**
Signed-off-by: Sylvain Rabot <s.rabot@lectra.com>

**Reason for Change**:

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: parts/k8s/cloud-init/jumpboxcustomdata.yml [hunk 1]
```diff
@@ -7,20 +7,20 @@ write_files:
   encoding: gzip
   owner: "root"
   content: !!binary |
-    {{CloudInitData "provisionSource"}}
+    {{ CloudInitData "provisionSource" }}
```
Comment (jackfrancis) [inline parts/k8s/cloud-init/jumpboxcustomdata.yml]: What's the purpose of adding extra whitespace after and before the `{{` and `}}`?

File: parts/k8s/cloud-init/nodecustomdata.yml [hunk 2]
```diff
@@ -6,135 +6,135 @@ write_files:
...
 - path: /opt/azure/containers/provision_cis.sh
   permissions: "0744"
   encoding: gzip
   owner: root
   content: !!binary |
-    {{CloudInitData "provisionCIS"}}
+    {{ CloudInitData "provisionCIS" }}
 
-{{if IsAzureStackCloud}}
+{{- if IsAzureStackCloud }}
```
Comment (jackfrancis) [inline parts/k8s/cloud-init/nodecustomdata.yml]: What's the purpose of this added `-` ?

File: parts/k8s/cloud-init/nodecustomdata.yml [hunk 3]
```diff
@@ -6,135 +6,135 @@ write_files:
...
 - path: /opt/azure/containers/provision_configs_custom_cloud.sh
   permissions: "0744"
   encoding: gzip
   owner: root
   content: !!binary |
-    {{WrapAsVariable "provisionConfigsCustomCloud"}}
-{{end}}
+    {{ WrapAsVariable "provisionConfigsCustomCloud" }}
+{{- end }}
```
Comment (jackfrancis) [inline parts/k8s/cloud-init/nodecustomdata.yml]: ditto

File: parts/k8s/cloud-init/nodecustomdata.yml (hunk 2 above)
Comment (sylr) [reply parts/k8s/cloud-init/nodecustomdata.yml]: The `-` tells go templates to discard any whitespaces located before `{{`.

It's useful to discard empty line generated when a line only contains a go template statement.

https://play.golang.org/p/6YdJA8vOQhe

File: parts/k8s/cloud-init/jumpboxcustomdata.yml (hunk 1 above)
Comment (sylr) [reply parts/k8s/cloud-init/jumpboxcustomdata.yml]: It's to be coherent with the `{{- ... -}}` syntax which can not suffer to not have whitespaces after or before the `-`.

File: parts/k8s/cloud-init/nodecustomdata.yml (hunk 2 above)
Comment (jackfrancis) [reply parts/k8s/cloud-init/nodecustomdata.yml]: Got it, thanks. Let's merge this in first:

https://github.com/Azure/aks-engine/pull/1684

That will probably cause merge conflicts with this changeset, FYI. So after we rebase and clear up conflicts, let's queue this up for E2E tests, review, and merge.

Thanks!

**Pull request closed without merging**